### PR TITLE
extract NEON_CHAIN_ID from elf symbols

### PR DESCRIPTION
**Pull request by ivandzen**
parse NEON_CHAIN_ID ELF symbol in proxy

### CODE REVIEW

File: proxy/plugin/solana_rest_api.py [hunk 1]
```diff
@@ -107,10 +106,14 @@ def web3_clientVersion(self):
...
-        return chainId
+        neon_config_load(self)
+        # NEON_CHAIN_ID is a string in decimal form
+        return hex(int(self.neon_config_dict['NEON_CHAIN_ID']))
 
     def net_version(self):
-        return str(int(chainId,base=16))
+        neon_config_load(self)
+        # NEON_CHAIN_ID is a string in decimal form
+        return self.neon_config_dict['NEON_CHAIN_ID']
```
Comment (vasiliy-zaznobin) [inline proxy/plugin/solana_rest_api.py:116]: Are network id and chain id the same thing?

File: proxy/plugin/solana_rest_api.py [hunk 2]
```diff
@@ -107,10 +106,14 @@ def web3_clientVersion(self):
         return self.neon_config_dict['web3_clientVersion']
 
     def eth_chainId(self):
-        return chainId
+        neon_config_load(self)
+        # NEON_CHAIN_ID is a string in decimal form
+        return hex(int(self.neon_config_dict['NEON_CHAIN_ID']))
```
Comment (vasiliy-zaznobin) [inline proxy/plugin/solana_rest_api.py:111]: Please add a test for eth_chainId

File: proxy/plugin/solana_rest_api.py (hunk 1 above)
Comment (ivandzen) [reply proxy/plugin/solana_rest_api.py:116]: For now - yes. It is the same number but in different formats: chain_id is 16-th base integer, net_version is 10-th base

File: proxy/plugin/solana_rest_api.py (hunk 2 above)
Comment (ivandzen) [reply proxy/plugin/solana_rest_api.py:111]: test added

**Review by vasiliy-zaznobin — APPROVED** [review]

**Pull request merged**